src/main/java/docSharing/controller/AbstractController.java

```diff
@@ -44,66 +58,117 @@ public ResponseEntity<List<FileRes>> getAll(Long parentFolderId, Long userId) {
         return ResponseEntity.ok().body(convertToFileRes(folders, documents));
     }
 
+    /**
+     * convertToFileRes is an inner function of getAll, that gets a list of folders & documents,
+     * and return a list of FileRes entity which has the name,id and the type of a given file
+     * for the convenient of the client side which need
+     * to show all the files to user.
+     *
+     * @param folders   - List<Folder> to present.
+     * @param documents - List<Document> to present
+     * @return - List<FileRes>
+     */
     private List<FileRes> convertToFileRes(List<Folder> folders, List<Document> documents) {
+        logger.info("in AbstractController -> convertToFileRes");
         List<FileRes> fileResList = new ArrayList<>();
         for (Folder folder :
                 folders) {
 
             fileResList.add(new FileRes(folder.getName(), folder.getId(), Type.FOLDER, Permission.ADMIN, folder.getUser().getEmail()));
         }
-        for (Document document :
-                documents) {
+        for (Document document : documents) {
             fileResList.add(new FileRes(document.getName(), document.getId(), Type.DOCUMENT, Permission.ADMIN, document.getUser().getEmail()));
         }
         return fileResList;
     }
 
-
+    /**
+     * create is a request from the client to create a new file in a specific folder location.
+     *
+     * @param item - item of kind folder or document.
+     * @param c    - the class of the item, need to know to what service sends the request.
+     * @return - ResponseEntity.
+     */
     public ResponseEntity<String> create(GeneralItem item, Class c) {
         // make sure we got all the data from the client
+        logger.info("in AbstractController -> create");
         try {
             Validations.validate(Regex.FILE_NAME.getRegex(), item.getName());
 //            Validations.validate(Regex.ID.getRegex(), item.getParentFolderId().toString());
         } catch (NullPointerException e) {
+            logger.error("in AbstractController -> create -> " + e.getMessage());
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
         } catch (IllegalArgumentException e) {
+            logger.error("in AbstractController -> create -> " + e.getMessage());
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
         }
-
         return ResponseEntity.ok().body(convertFromClassToService(c).create(item).toString());
     }
 
+    /**
+     * rename a file, called from the fileController with a request to change name.
+     *
+     * @param id   - of a file.
+     * @param name - new name.
+     * @param c    - the class of the item, need to know to what service sends the request.
+     * @return - ResponseEntity.
+     */
     public ResponseEntity<Object> rename(Long id, String name, Class c) {
-        if (name == null)
+        logger.info("in AbstractController -> rename");
+        if (name == null) {
```

Comment:
This wasn't your piece of code, but we already have a validation class, so you can use it instead of always checking if something is null (since the validation class does it by itself)
Validations.validate() function works for it with the proper REGEX enum.
This is of course right about all the ==null validations